lib/merge.rb

```diff
@@ -0,0 +1,30 @@
+class Array
+  def merge_sort
+    if self.length <= 1
+      return self
+    end
+
+    middle = (self.length / 2).floor
+    left = self.slice(0...middle).merge_sort
+    right = self.slice(middle...self.length).merge_sort
+
+    sorted = []
+
+    while (left.length > 0) || (right.length > 0)
+      if left.length == 0
+        sorted += right
+        return sorted
+      elsif right.length == 0
+        sorted += left
+        return sorted
```

Comment:
This `return`, and the one on line 16, are unneeded, eh? The next line executed after these, will return `sorted` anyway?